sdk/communication/azure-communication-chat/azure/communication/chat/_shared/models.py

```diff
@@ -70,9 +87,15 @@ def __init__(self, id, **kwargs):
         # type: (str, Any) -> None
         self.raw_id = kwargs.get('raw_id', id)
         self.properties = CommunicationUserProperties(id=id)
+        if self.raw_id is None:
+            self.raw_id = _communication_user_raw_id(self)
 
     def __eq__(self, other):
-        return self.__dict__ == other.__dict__
+        return self.raw_id == _communication_user_raw_id(other)
```

Comment:
why can not we directly check against `other.raw_id` instead of getting it from identifier properties? same comment for other identifiers

Reply:
in python it is public and the `_communication_user_raw_id` by prefix underscore we are tagging as it is used internally only. so here we can get the raw id directly instead of calling internal method, I am not convinced at all but ok by me using it either way :)